packages/flutter/lib/src/material/drawer.dart

```diff
@@ -218,34 +235,72 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
 
   void _move(DragUpdateDetails details) {
     final double delta = details.primaryDelta / _width;
-    switch (Directionality.of(context)) {
-      case TextDirection.rtl:
-        _controller.value -= delta;
-        break;
-      case TextDirection.ltr:
-        _controller.value += delta;
-        break;
+    if (widget.type == DrawerType.START) {
```

Comment:
Rather than break down these case by `DrawerAlignment`.  We should translate the visual `delta` into a logical `valueDelta` by (possibly) negating it once for `DrawerAlignment` and (possibly) again for `Directionality`.  Then we can apply the `valueDelta` to `_controller.value` on a single line of code.

Reply:
Fixed